docs/tutorials/data_manipulation/dataframesdataset.md

```diff
@@ -0,0 +1,375 @@
+categorical features are provided, DeepAR also needs the cardinality as input.
+We have one static categorical feature that can take on two
+values (0 or 1). The cardinality in this case is a list of one element `[2,]`.
+
+
+```python
+def train_and_predict_with_features(ds, prediction_length):
+    model = DeepAREstimator(
+        freq=ds.freq,
+        prediction_length=prediction_length,
+        use_feat_dynamic_real=True,
+        use_feat_static_cat=True,
+        cardinality=[2,],
+        trainer=Trainer(epochs=1),
+    )
+    predictor = model.train(ds)
+    forecast_it, ts_it = make_evaluation_predictions(dataset=ds, predictor=predictor)
+    evaluator = Evaluator(quantiles=(np.arange(20) / 20.0)[1:])
+    agg_metrics, item_metrics = evaluator(ts_it, forecast_it, num_series=len(ds))
+    return agg_metrics["MSE"]
+```
+
+Lets now generate single time series and multiple time series
+(as list and as long-dataframe).
+Note we don't provide `freq` and `timestamp`, because they are automatically
+inferred from the index. We are also not passing "target" and "item_id"
+(for long-dataframe) because those are the default values of `target`
+and `item_id`, respectively.
+
+
+```python
+single_ts = generate_single_ts_with_features(date_range, item_id=0)
+multiple_ts = [generate_single_ts_with_features(date_range, item_id=i) for i in range(N)]
+multiple_ts_long = pd.concat(multiple_ts)
+
+single_ts_dataset = DataFramesDataset(
+    single_ts,
+    feat_dynamic_real=["dynamic_real_1", "dynamic_real_2"],
+    feat_static_cat=["static_cat_1"],
+)
+multiple_ts_dataset = DataFramesDataset(
+    multiple_ts,
+    feat_dynamic_real=["dynamic_real_1", "dynamic_real_2"],
+    feat_static_cat=["static_cat_1"],
+)
+multiple_ts_dataset_dict = DataFramesDataset(
+    {i: _ts for i, _ts in enumerate(multiple_ts)},
+    feat_dynamic_real=["dynamic_real_1", "dynamic_real_2"],
+    feat_static_cat=["static_cat_1"],
+)
+# for long-dataset we use a different constructor and need a `item_id` column
+multiple_ts_long_dataset = DataFramesDataset.from_long_dataframe(
+    multiple_ts_long,
+    feat_dynamic_real=["dynamic_real_1", "dynamic_real_2"],
+    feat_static_cat=["static_cat_1"],
+)
+```
+
+Thats it! We can now call `train_and_predict_with_features` with all the datasets.
+
+
+```python
+train_and_predict_with_features(single_ts_dataset, prediction_length)
+```
+
+```python
+train_and_predict_with_features(multiple_ts_dataset, prediction_length)
+```
+
+```python
+train_and_predict_with_features(multiple_ts_dataset_dict, prediction_length)
+```
+
+```python
+train_and_predict_with_features(multiple_ts_long_dataset, prediction_length)
+```
+
+## Use train/test split
+
+Here, we split the DataFrame/DataFrames into training and test data
+using the `OffsetSplitter` from gluonts. We can then
+use the training data to train the model and the test data for prediction.
+Note, the `OffsetSplitter` needs the `item_id`. If you want to split
```

Comment:
I think this detail of splitters (as well as other details) will change soon (see #2031, #2041). Maybe it's better to not "commit" too much to their current interface, and instead rely directly on slicing for splitting here? After all, one reason to support `pd.DataFrame` as datasets is that people can manipulate them easily in Pandas-land.

Something like

```python
train = DataFramesDataset(df[:-prediction_length])
test = DataFramesDataset(df)
```

should do it.